src/scripts/find_dadis_transboundary_ids.py

```diff
@@ -0,0 +1,221 @@
+    df = df.merge(species_df, how="left", on="dadis_species_id")
+    df = df.rename(columns={"id": "dadis_transboundary_id", "name": "dadis_breed_name"})
+    return df
+
+
+def get_all_dadis_transboundary(client: DadisClient) -> pd.DataFrame:
+    """
+    Get all names for DADIS transboundary breeds, some VBO entries
+    may use non-canonical names
+    """
+    resp = client.get_all_transboundary_breeds()
+    df = (
+        pd.DataFrame.from_records([b.model_dump() for b in resp.response])
+        .rename(
+            columns={
+                "speciesId": "dadis_species_id",
+                "name": "dadis_breed_name",
+                "transboundaryId": "dadis_transboundary_id",
+                "id": "dadis_breed_id",
+                "iso3": "dadis_iso3_code",
+            }
+        )
+        .drop_duplicates(
+            subset=["dadis_species_id", "dadis_breed_name", "dadis_transboundary_id"]
+        )
+    )
+    species_df = get_dadis_species(client)
+    result = df.merge(species_df, how="left", on="dadis_species_id").sort_values(
+        ["dadis_transboundary_id", "dadis_breed_name"]
+    )
+    return result
+
+
+def get_simple_matches(vbo_data: pd.DataFrame, client: DadisClient) -> pd.DataFrame:
+    """
+    Match VBO entries to DADIS transboundary breeds based on their
+    canonical names. Return a dataframe containing the matches
+    """
+    dadis_canonical = get_canonical_dadis_transboundary(client=client)
+    simple_matches = (
+        vbo_data[["VBO id", "term label", "DADIS_name", "DADIS_species_name"]]
+        .merge(
+            dadis_canonical,
+            how="left",
+            left_on=["DADIS_name", "DADIS_species_name"],
+            right_on=["dadis_breed_name", "species_name"],
+            sort=False,
+        )
+        .drop(columns=["dadis_breed_name", "species_name"])
+        .convert_dtypes()
+    )
+    return simple_matches
+
+
+def get_extra_matches(vbo_data: pd.DataFrame, client: DadisClient) -> pd.DataFrame:
+    dadis_all = get_all_dadis_transboundary(client=client)
+    extra_matches = (
+        vbo_data[["VBO id", "term label", "DADIS_name", "DADIS_species_name"]]
+        .merge(
+            dadis_all,
+            how="left",
+            left_on=["DADIS_name", "DADIS_species_name"],
+            right_on=["dadis_breed_name", "species_name"],
+            indicator=True,
+            # Need to ensure we use sort=False so order stays consistent with original
+            sort=False,
+        )
+        .drop(columns=["dadis_breed_name", "species_name"])
+        .convert_dtypes()
+        .drop_duplicates(subset=["VBO id", "dadis_transboundary_id"])
+    )
+    counts = extra_matches["VBO id"].value_counts()
+    duplicates = counts.loc[counts >= 2].index.tolist()
+    multiple_matches = extra_matches.loc[extra_matches["VBO id"].isin(duplicates), :]
+    print(
+        "The following entries matched against multiple DADIS entries - will not be updated"
+    )
```

Comment:
Did this ever happen to you (apart from the pony)?

I would prefer we raise an exception here instead of a print, but for sure use `logging.warning()`.

Reply:
This is a different issue to the pony, where the spreadsheet from VBO had two rows for VBO:0000991, there was only one VBO entry with that problem.

These are entries where we can't uniquely match to DADIS based on the breed name and species, e.g. there are two different DADIS entries with the breed name "Native of North Africa", e.g. 
 
![Screenshot 2023-10-23 at 10 09 26 am](https://github.com/monarch-initiative/vertebrate-breed-ontology/assets/61438033/4ec6e605-dbfe-42ad-8ff0-ba5c8bd805ad)

There are only about 20 VBO entries with this issue, so they can probably be fixed manually - I can provide the spreadsheet listing them.

Reply:
So there are 20 VBO entries, which are currently correspond to more than one breed/country/species combination in DADIS. So lets say we have VBO:123, it is possible that it corresponds to DADIS:321 and DADIS:322 (two distinct records in the DADIS database.

In this case, what is the differentiating factor? I thought we were told the country/breed/species constitutes a primary key?